library/cisco_imc_server.py

```diff
@@ -0,0 +1,211 @@
+    from imcsdk.apis.server.serveractions import server_power_down
+    from imcsdk.apis.server.serveractions import server_power_down_gracefully
+    from imcsdk.apis.server.serveractions import server_power_cycle
+
+    ansible = module.params
+    timeout, interval = ansible["timeout"], ansible["interval"]
+    server_id = ansible["server_id"]
+    changed = False
+
+    current = server_power_state_get(server, server_id=server_id)
+
+    if current == "off":
+        if status == "on" or status == "boot":
+            changed = server_power_up(server, timeout=timeout, 
+                            interval=interval, serverid_=server_id)
+        
+    elif current == "on":
+        if status == "shutdown":
+            changed = server_power_down_gracefully(server, timeout=timeout, 
+                            interval=interval, serverid_=server_id)
+        elif status == "off":
+            changed = server_power_down(server, timeout=timeout, 
+                            interval=interval, serverid_=server_id)
+        elif status == "boot" or status == "reset":
+             changed = server_power_cycle(server, timeout=timeout, 
+                            interval=interval, serverid_=server_id)
+            
+    return change
+
+
+def setup_server_led(server, module, locator_led):
+    from imcsdk.apis.server.serveractions import locator_led_off
+    from imcsdk.apis.server.serveractions import locator_led_on
+
+    server_id, chassis_id = ansible["server_id"], ansible["chassis_id"]
+    
+    if locator_led == "on": 
+        locator_led_on(server,
+                       server_id=server_id,
+                       chassis_id=chassis_id)
+    elif locator_led == "off":
+        locator_led_off(server, 
+                        server_id=server_id, 
+                        chassis_id=chassis_id)
+    return True
+
+    
+
+def setup_server(server, module):
+    ansible = module.params
+    status, locator_led = ansible["status"], ansible["locator_led"]
+    
+    if status is not None:
+        power_changed = setup_server_power(server, module, 
+                               status, timeout, interval)
+        
+    if locator_led is not None:
+        led_changed = setup_server_led(server, module, locator_led)
+            
+    return (power_changed or led_changed)
+
+
+def setup(server, module):
+    results = {}
+    err = False
+
+    try:
+        results["changed"] = setup_server(server, module)
+
+    except Exception as e:
+        err = True
+        results["msg"] = "setup error: %s " % str(e)
+        results["changed"] = False
+
+    return results, err
+
+
+def main():
+    from ansible.module_utils.cisco_imc import ImcConnection
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_id=dict(required=False, type='int', default=1),
+            chassis_id=dict(required=False, type='int', default=1),
+            state=dict(required=False, type='str',
```

Comment:
I don't' believe so.  status implies "position of affairs" .. state implies "particular condition at a time". Most of the ansible modules use 'state' as the moniker: file, service, apt, yum. The only one that uses 'status' AFAIK is the return value of 'stat'.